### PR TITLE
Splitting PackagesBuilder with a new PackageSelection Object

### PR DESCRIPTION
**Pull request by JamesRezo**
following up the goal of #253 :
+ it splits the business logics of package selection and packages build
+ Unit Tests

### CODE REVIEW

**Comment by alcohol** [comment]
If you could add a `.travis.yml` that would be great :-)

Initially you can look at the one [composer/semver](/composer/semver) or [composer/spdx-license](/composer/spdx-licenses) use.

**Comment by JamesRezo** [comment]
Done ! :)

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+language: php
+
+sudo: false
+
+cache:
+  directories:
+    - $HOME/.composer/cache
```
Comment (stof) [inline .travis.yml]: I suggest caching only the ``/files`` subfolder. See https://github.com/puli/manager/pull/9#discussion_r40499480 for the explanation

**Comment by alcohol** [comment]
Thanks :)

@stof you recommend I also modify this caching path on the other composer repos?

**Pull request merged**